src/api/ApiClient.js

```diff
@@ -81,9 +95,19 @@ export default class ApiClient {
       }
     }
 
-    return fetch(`${this.prefix}/${url}`, init).then(res => res.json())
+    return fetch(`${this.prefix}/${url}`, init).then((res) => {
+      ({ status } = res);
+      return res.json();
+    })
       .then((data) => {
         if (status >= 400 && status !== 422) {
+          if (status === 401 && data.message === 'Not Authorized') {
```

Comment:
next `if` statement also checks is status === 401 or not. Probably you need reorganize the logic